addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SiteIsolationScanRule.java

```diff
@@ -0,0 +1,265 @@
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/**
+ * Spectre vulnerability has shown that Javascript code can be used to read any part of memory in
+ * the same address space. Browser architectures are been <a href=
+ * "https://chromium.googlesource.com/chromium/src/+/master/docs/security/side-channel-threat-model.md">re-think</a>
+ * to keep sensitive data outside of the address space of untrusted code.
+ *
+ * <p>To achieve this, three headers have been added:
+ *
+ * <ul>
+ *   <li>Cross-Origin-Resource-Policy:
+ *   <li>Cross-Origin-Embedder-Policy: only allow resources that have enabled CORP ou CORS
+ *   <li>Cross-Origin-Opener-Policy: allow sites to control browsing context group
+ * </ul>
+ *
+ * @see <a
+ *     href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Cross-Origin_Resource_Policy_(CORP)">CORP
+ *     on MDN</a>
+ * @see <a
+ *     href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy">COEP
+ *     on MDN</a>
+ * @see <a
+ *     href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy">COOP
+ *     on MDN</a>
+ * @see <a href="https://fetch.spec.whatwg.org/#cross-origin-resource-policy-header">COOP Specs</a>
+ * @see <a href="https://html.spec.whatwg.org/multipage/origin.html#coep">COEP Specs</a>
+ */
+public class SiteIsolationScanRule extends PluginPassiveScanner {
+    /** Prefix for internationalized messages used by this rule */
+    private static final String SITE_ISOLATION_MESSAGE_PREFIX = "pscanalpha.site-isolation.";
+
+    private static final int PLUGIN_ID = 90004;
+
+    private final SiteIsolationHeaderScanner[] scanners =
+            new SiteIsolationHeaderScanner[] {
+                new CorpHeaderScanner(this::newAlert),
+                new CoepHeaderScanner(this::newAlert),
+                new CoopHeaderScanner(this::newAlert)
+            };
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+
+        // Specs don't state that errors pages should be excluded
+        // However, successful responses are associated to a resource
+        // that should be protected.
+        // Only consider HTTP Status code 2XX to avoid a False Positive
+        if (!HttpStatusCode.isSuccess(msg.getResponseHeader().getStatusCode())) {
+            return;
+        }
+
+        Stream.of(scanners).forEach(s -> s.check(msg.getResponseHeader()));
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        // Nothing to do.
+    }
+
+    @Override
+    public String getName() {
+        return SITE_ISOLATION_MESSAGE_PREFIX + "name";
```

Comment:
```suggestion
        return Constant.messages.getString(SITE_ISOLATION_MESSAGE_PREFIX + "name");
```

To address the build (test) failure.